client/extensions/woocommerce/app/settings/email/mailchimp/index.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { filter, matches } from 'lodash';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import MailChimpGettingStarted from './getting-started';
+import MailChimpSetup from './setup-mailchimp';
+import MailChimpDashboard from './mailchimp_dashboard';
+import QueryJetpackPlugins from 'components/data/query-jetpack-plugins';
+import QueryMailChimpSettings from 'woocommerce/state/sites/settings/email/querySettings';
+import { isRequestingForSites } from 'state/plugins/installed/selectors';
+import { getPlugins } from 'state/plugins/installed/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { localize } from 'i18n-calypso';
+import { mailchimpSettings, isRequestingSettings } from 'woocommerce/state/sites/settings/email/selectors';
+import Card from 'components/card';
+
+class MailChimp extends React.Component {
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			setupWizardStarted: false
+		};
+	}
+
+	startWizard = () => {
+		this.setState( { setupWizardStarted: true } );
+	}
+
+	closeWizard = () => {
+		this.setState( { setupWizardStarted: false } );
+	}
+
+	render() {
+		const { siteId, isRequestingMailChimpSettings, isRequestingPlugins } = this.props;
+		const className = classNames( 'mailchimp__main', { mailchimp__loading: isRequestingMailChimpSettings } );
+		const { setupWizardStarted } = this.state;
+		const isRequestingData = ( isRequestingMailChimpSettings || isRequestingPlugins );
+		const mailChimpIsReady = ! isRequestingData && ( this.props.settings.active_tab === 'sync' );
+		return (
+			<div className={ className }>
+				<QueryJetpackPlugins siteIds={ [ siteId ] } />
+				<QueryMailChimpSettings siteId={ siteId } />
+				{ isRequestingData && <Card> Mailchimp is Loading </Card> }
+				{ mailChimpIsReady && <MailChimpDashboard siteId={ siteId } onClick={ this.startWizard } /> }
+				{ ! setupWizardStarted && ! isRequestingData && this.props.settings.active_tab !== 'sync' &&
```

Comment:
again if `settings` was pulled out of props above this could be shorted, and could it be null?